File: code/typescript/builder/Director.ts
import { IBuilder } from "./IBuilder";

export class Director {
    private builder: IBuilder;

    constructor(builder: IBuilder) {
        this.builder = builder;
    }

    public constructSandwich(): void {
        this.builder.buildPartA();
        this.builder.buildPartB();
        this.builder.buildPartC();
    }
}

File: code/csharp/DesignPatterns/builder/Director.cs
public class Director
{
    private IBuilder _builder;

    public Director(IBuilder builder)
    {
        _builder = builder;
    }

    public void ConstructSandwich()
    {
        _builder.BuildPartA();
        _builder.BuildPartB();
        _builder.BuildPartC();
    }
}
